File: app/KMLViewController.m
/* Copyright (c) 2016 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#if !defined(__has_feature) || !__has_feature(objc_arc)
#error "This file requires ARC support."
#endif

#import "KMLViewController.h"

#import <GoogleMaps/GoogleMaps.h>

#import "GMUKMLParser.h"
#import "GMUGeometryRenderer.h"

static const double kCameraLatitude = 37.4220;
static const double kCameraLongitude = -122.0841;

@implementation KMLViewController {
  GMSMapView *_mapView;
}

- (void)loadView {
  GMSCameraPosition *camera =
      [GMSCameraPosition cameraWithLatitude:kCameraLatitude longitude:kCameraLongitude zoom:17];
  _mapView = [GMSMapView mapWithFrame:CGRectZero camera:camera];
  self.view = _mapView;
}

- (void)viewDidLoad {
  [super viewDidLoad];
  NSString *path = [[NSBundle mainBundle] pathForResource:@"KML_Sample" ofType:@"kml"];
  NSURL *url = [NSURL fileURLWithPath:path];
  GMUKMLParser *parser = [[GMUKMLParser alloc] initWithURL:url];
  [parser parse];
  GMUGeometryRenderer *renderer = [[GMUGeometryRenderer alloc] initWithMap:_mapView
                                                                geometries:parser.placemarks
                                                                    styles:parser.styles];
  [renderer render];
}

@end

File: samples/SwiftDemoApp/SwiftDemoApp/KMLViewController.swift
/* Copyright (c) 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import GoogleMaps
import UIKit
import GoogleMapsUtils

class KMLViewController: UIViewController {
  private var mapView: GMSMapView!
  private var renderer: GMUGeometryRenderer!
  private var kmlParser: GMUKMLParser!
  private var geoJsonParser: GMUGeoJSONParser!

  override func loadView() {
    let camera = GMSCameraPosition.camera(withLatitude: 37.4220, longitude: -122.0841, zoom: 17)
    mapView = GMSMapView.map(withFrame: CGRect.zero, camera: camera)
    self.view = mapView

    let path = Bundle.main.path(forResource: "KML_Sample", ofType: "kml")
    let url = URL(fileURLWithPath: path!)
    kmlParser = GMUKMLParser(url: url)
    kmlParser.parse()

    renderer = GMUGeometryRenderer(map: mapView,
                                   geometries: kmlParser.placemarks,
                                   styles: kmlParser.styles,
                                   styleMaps: kmlParser.styleMaps)

    renderer.render()
  }
}
